crates/wasm-mutate/src/mutators/peephole/eggsy/mod.rs

```diff
@@ -0,0 +1,691 @@
+        match eterm {
+            Lang::I32Mul(_) => {
+                newfunc.instruction(Instruction::I32Mul);
+            }
+            Lang::I32Const(v) => {
+                newfunc.instruction(Instruction::I32Const(*v));
+            }
+            Lang::I32Shl(_) => {
+                newfunc.instruction(Instruction::I32Shl);
+            }
+            Lang::I32Add(_) => {
+                newfunc.instruction(Instruction::I32Add);
+            }
+            Lang::I32Sub(_) => {
+                newfunc.instruction(Instruction::I32Sub);
+            }
+            Lang::I32And(_) => {
+                newfunc.instruction(Instruction::I32And);
+            }
+            Lang::I32Or(_) => {
+                newfunc.instruction(Instruction::I32Or);
+            }
+            Lang::I32Xor(_) => {
+                newfunc.instruction(Instruction::I32Xor);
+            }
+            Lang::I32ShrU(_) => {
+                newfunc.instruction(Instruction::I32ShrU);
+            }
+            Lang::I32Popcnt(_) => {
+                newfunc.instruction(Instruction::I32Popcnt);
+            }
+            Lang::ILoad(_) => {
+                // Do nothing
+                let operators = &operators[insertion_point..=insertion_point + 1/* take to the next operator to save the offset */];
+                let range = (operators[0].1, operators[1].1);
+
+                // Copy the mem operation
+                let raw_data = &info.get_code_section().data[range.0..range.1];
+                newfunc.raw(raw_data.iter().copied());
+            }
+            Lang::Rand => {
+                newfunc.instruction(Instruction::I32Const(rnd.gen()));
+            }
+            Lang::Unfold(_) => {
+                let child = operands[current][0];
+                let node = id_to_node[usize::from(child)];
+
+                match node {
+                    Lang::I32Const(value) => {
+                        let r: i32 = rnd.gen();
+
+                        newfunc.instruction(Instruction::I32Const(r));
+                        newfunc.instruction(Instruction::I32Const((Wrapping(r) - Wrapping(*value)).0));
+                        newfunc.instruction(Instruction::I32Add);
+                    },
+                    _ => panic!("The oeprand for this operator should be a constant, check if the rewriting rule is defined with such conditions")
+                }
+            }
+            Lang::Symbol(s) => {
+                // Copy the byte stream to aavoid mapping
+                let entryidx = symbolsmap[&s.to_string()];
+                let entry = &minidfg.entries[entryidx];
+
+                // Write the symbol in the correct place of the functions
+
+                let bytes = &info.get_code_section().data
+                    [entry.byte_stream_range.start..entry.byte_stream_range.end];
+                newfunc.raw(bytes.iter().copied());
+                //newfunc.instruction(Instruction::I32Const(*v));
+            }
```

Comment:
I see your point. I will focus in this issue ASAP. 

In fact there is a bug in line 256. What it should  happen is that instead of writing the symbol we should send the entry back to the writing since a symbol can be a `symbolized` part of the dfg, I will fix it.

Other thing here, what is happening right now is that we are writing down de dfg roots in the order they appear using the `parents` field in the `MiniDFG` struct. 

For example in the code above, we will generate the following `parents` relation:

```wasm
;0; call $get_a                    ;(parent 1);
;1; i32.load                         ;(parent 7) this is a root in the dfg;
;2; call $get_b                   ;(parent 3);
;3; i32.load                        ;(parent 7);
;4; i32.const 0x12345678 ;(parent 5);
;5; i32.load                         ;(parent 6);
;6; drop                              ;(parent -1);
;7; i32.add                          ;(parent -1) (child: 3 1);
```
Our algorithm right now will write in order then the dfgs starting at 6 and 7. If we imagine for example applying the commutative rewriting in 7 we will have then.

```wasm
i32.const 0x12345678
i32.load
drop ;root 6;
call $get_b
i32.load
call $get_a
i32.load
i32.add ; root 7;
```

The problem that I am seeing with our current implementation approach is that the operands will be written near to the operator and not in the corresponding places even when the stack-neutral is maintained. The ideal scenario is if 3 and 1 could be replaced in their positions. To my understanding this could be more expensive. 

The thing is that since we are constructing the dfg, I am not seeing the locals usage a very clean solution, but it seems the only one we have now :)